### DISCUSSION TITLE
crypto.sha256 (or 512 for that matter) hash files

### DISCUSSION THREAD

**Discussion by danabooth** [vlang/v#21642] [Questions and Answers] [answered]
Is there a quick and easy way to hash a file without having to manually open the stream? I know that some other languages have this, but I didn't really find anything in the V docs about this.

**Comment by JalonSolov** [comment]
Those other languages are just adding "sugar" - the stream still has to be opened, their lib routines just do it for you.

**Reply by danabooth** [reply]
Yes, I know, I just didn't want to do it by hand just to have some stackoverflow smart a$$ condescend to me at some point.

**Reply by JalonSolov** [reply, marked as answer]
V doesn't have it now, but PRs are always welcome.  :-)

One example of doing what you want (plus a whole lot more) can be found in the coreutils code.

https://github.com/vlang/coreutils/blob/main/common/sums/sums.v

That's the common code for the `sha*sum` tools.

**Reply by danabooth** [reply]
Thanks, that's good stuff.
